### PR TITLE
Fix cat recovery display of bytes fields

### LINKED ISSUE #40335: Why doesn’t _cat/recovery honor the bytes=XX in the URI?
**Describe the feature**:

**Elasticsearch version** (`bin/elasticsearch --version`):

```
$ rpm -qi elasticsearch | grep Vers
Version     : 6.5.4
```

**Plugins installed**: []

**JVM version** (`java -version`):

```
$ java -version
openjdk version "1.8.0_191"
OpenJDK Runtime Environment (build 1.8.0_191-b12)
OpenJDK 64-Bit Server VM (build 25.191-b12, mixed mode)
```

**OS version** (`uname -a` if on a Unix-like system):

```
$ cat /etc/redhat-release
CentOS Linux release 7.5.1804 (Core)
```

**Description of the problem including expected versus actual behavior**:

**Steps to reproduce**:

I'm trying to change the units for the following API call:

```
GET /_cat/recovery?bytes=gb
```

The docs around the cat API imply that you can do this, but the recovery endpoint doesn't seem to work with this. The units are always shown in bytes.

Example:

```
GET '_cat/recovery?bytes=gb&v&h=index,shard,time,type,stage,source_node,target_node,files,files_recovered,files_percent,bytes_total,bytes_percent' | grep -v done | head
index                             shard time  type           stage source_node     target_node     files files_recovered files_percent bytes_total  bytes_percent
filebeat-6.5.1-2019.03.06         0     2.7h  peer           index rdu-es-data-01e rdu-es-data-01b 268   262             97.8%         218212992946 85.2%
filebeat-6.5.1-2019.03.06         1     2.7h  peer           index rdu-es-data-01a rdu-es-data-01e 277   276             99.6%         228721023484 98.6%
filebeat-6.5.1-2019.03.05         1     2.1h  peer           index rdu-es-data-01d rdu-es-data-01b 271   262             96.7%         203869559392 77.1%
```

REF: https://www.elastic.co/guide/en/elasticsearch/reference/current/cat.html 1

I asked this Q over on the forums and it was mentioned that this appears to be a bug, specifically here in the code - https://github.com/elastic/elasticsearch/blob/99f88f15c5febbca2d13b5b5fda27b844153bf1a/server/src/main/java/org/elasticsearch/rest/action/cat/RestRecoveryAction.java#L167-L170.

Link to the forum Q: https://discuss.elastic.co/t/why-doesnt-cat-recovery-honor-the-bytes-xx-in-the-uri/173241/2

> These values are returned as numbers, not as ByteSizeValues, so the ?bytes= parameter doesn't apply. Could you open an issue on Github, @slmingol?

**Provide logs (if relevant)**:

### PR DESCRIPTION
**Pull request by jasontedor**
The cat APIs support the ?bytes parameter to format bytes value. The cat recovery API does not follow this because the fields were not using ByteSizeValue. This commit addresses this.

Closes #40335

### CODE REVIEW

File: rest-api-spec/src/main/resources/rest-api-spec/test/cat.recovery/10_basic.yml [hunk 1]
```diff
@@ -122,10 +122,10 @@
           \d+         \s+                                 # files_recovered
           \d+\.\d+%   \s+                                 # files_percent
           \d+         \s+                                 # files_total
-          \d+         \s+                                 # bytes
-          \d+         \s+                                 # bytes_recovered
+          \d+(b|kb|mb|gb|tb|pb) \s+                                 # bytes
+          \d+(b|kb|mb|gb|tb|pb) \s+                                 # bytes_recovered
```
Comment (dnhatn) [inline rest-api-spec/src/main/resources/rest-api-spec/test/cat.recovery/10_basic.yml]: nit: indentation

File: rest-api-spec/src/main/resources/rest-api-spec/test/cat.recovery/10_basic.yml [hunk 2]
```diff
@@ -35,10 +35,10 @@
                 \d+         \s+                                 # files_recovered
                 \d+\.\d+%   \s+                                 # files_percent
                 \d+         \s+                                 # files_total
-                \d+         \s+                                 # bytes
-                \d+         \s+                                 # bytes_recovered
+                \d+(b|kb|mb|gb|tb|pb) \s+                                 # bytes
+                \d+(b|kb|mb|gb|tb|pb) \s+                                 # bytes_recovered
```
Comment (dnhatn) [inline rest-api-spec/src/main/resources/rest-api-spec/test/cat.recovery/10_basic.yml]: nit: indentation

**Review by dnhatn — APPROVED** [review]
LGTM. I left two small nits on indentation.

**Comment by jasontedor** [comment]
@elasticmachine run elasticsearch-ci/1
@elasticmachine run elasticsearch-ci/oss-distro-docs

**Comment by jasontedor** [comment]
@elasticmachine run elasticsearch-ci/2

**Comment by DaveCTurner** [comment]
Copying my thoughts here so they're not lost:

I do think that this is a bug, but I think there's a risk that it's a bug that users rely on. This change will break users of this API, e.g for monitoring recovery progress. I also think that the `bytes` name for these fields doesn't fit well if the units are added like this. I think we should avoid this breaking change and instead:

- add fields `size_to_recover`, `size_recovered` and `size_total` (also `size_percent` for consistency) with these values in 7.x, but do not add them to the default format.
- deprecate the `bytes_*` fields in 7.x but do not remove them from the default format.
- remove the `bytes_*` fields in 8.0 and replace them with the `size_*` fields in the default format.

I think this is sufficient to avoid a breaking change without needing to do too much of a deprecation dance: this would mean that the only way to avoid a deprecation warning in 7.x would be to pass the `?h=...` parameter and specify the desired fields, and for none of those fields to be a `bytes_*` field. These warning-free requests would be unbroken by the change in the default format in 8.0.

**Comment by jasontedor** [comment]
@DaveCTurner I would like to avoid changing the names as I think that's a breaking change with larger impact, even if we do a deprecation dance. How would you feel about integrating this to 8.0.0 only?

**Comment by DaveCTurner** [comment]
Yes, I'm ok with this going into 8.0 only.

**Pull request merged**